provision.py

```diff
@@ -110,6 +110,21 @@
 
 LOUD = dict(_out=sys.stdout, _err=sys.stderr)
 
+def setup_node_modules():
+    output = subprocess.check_output(['sha1sum', 'package.json'])
+    sha1sum = output.split()[0]
+    success_stamp = os.path.join(os.environ['HOME'], 'success-stamp', sha1sum)
```

Comment:
Probably should come up with a better name for this, and also one compartmentalized to a particular `zulip` checkout; and also we probably want it to go away if someone does `rm -rf node_modules`.  Maybe `node_modules/.npm-success-stamp/` for the directory?